drivers/java/driver.go

```diff
@@ -0,0 +1,564 @@
+	}
+
+	// pluginInfo is the response returned for the PluginInfo RPC
+	pluginInfo = &base.PluginInfoResponse{
+		Type:             base.PluginTypeDriver,
+		PluginApiVersion: "0.0.1",
+		PluginVersion:    "0.1.0",
+		Name:             pluginName,
+	}
+
+	// configSpec is the hcl specification returned by the ConfigSchema RPC
+	configSpec = hclspec.NewObject(map[string]*hclspec.Spec{})
+
+	// taskConfigSpec is the hcl specification for the driver config section of
+	// a taskConfig within a job. It is returned in the TaskConfigSchema RPC
+	taskConfigSpec = hclspec.NewObject(map[string]*hclspec.Spec{
+		// It's required for either `class` or `jar_path` to be set,
+		// but that's not expressable in hclspec.  Marking both as optional
+		// and setting checking explicitly later
+		"class":     hclspec.NewAttr("class", "string", false),
+		"classpath": hclspec.NewAttr("classpath", "string", false),
+		"jar_path":  hclspec.NewAttr("jar_path", "string", false),
+		"java_opts": hclspec.NewAttr("java_opts", "list(string)", false),
+		"args":      hclspec.NewAttr("args", "list(string)", false),
+	})
+
+	// capabilities is returned by the Capabilities RPC and indicates what
+	// optional features this driver supports
+	capabilities = &drivers.Capabilities{
+		SendSignals: false,
+		Exec:        false,
+		FSIsolation: cstructs.FSIsolationNone,
+	}
+
+	_ drivers.DriverPlugin = (*Driver)(nil)
+)
+
+func init() {
+	if runtime.GOOS == "linux" {
+		capabilities.FSIsolation = cstructs.FSIsolationChroot
+	}
+}
+
+// TaskConfig is the driver configuration of a taskConfig within a job
+type TaskConfig struct {
+	Class     string   `codec:"class"`
+	ClassPath string   `codec:"classpath"`
+	JarPath   string   `codec:"jar_path"`
+	JvmOpts   []string `codec:"java_opts"`
+	Args      []string `codec:"args"` // extra arguments to java executable
+}
+
+// TaskState is the state which is encoded in the handle returned in
+// StartTask. This information is needed to rebuild the taskConfig state and handler
+// during recovery.
+type TaskState struct {
+	ReattachConfig *utils.ReattachConfig
+	TaskConfig     *drivers.TaskConfig
+	Pid            int
+	StartedAt      time.Time
+}
+
+// Driver is a driver for running images via Java
+type Driver struct {
+	// eventer is used to handle multiplexing of TaskEvents calls such that an
+	// event can be broadcast to all callers
+	eventer *eventer.Eventer
+
+	// tasks is the in memory datastore mapping taskIDs to taskHandle
+	tasks *taskStore
+
+	// ctx is the context for the driver. It is passed to other subsystems to
+	// coordinate shutdown
+	ctx context.Context
+
+	// nomadConf is the client agent's configuration
+	nomadConfig *base.ClientDriverConfig
+
+	// signalShutdown is called when the driver is shutting down and cancels the
+	// ctx passed to any subsystems
+	signalShutdown context.CancelFunc
+
+	// logger will log to the plugin output which is usually an 'executor.out'
+	// file located in the root of the TaskDir
```

Comment:
Comment was wrong in other drivers. Should be right on master (this logger logs back to the nomad agent's logs)